dciv_bot/module/misc.py

```diff
@@ -579,6 +580,85 @@ async def invite(self, ctx):
+            - A dice roll can be followed by an Ln or Hn, where it will discard the lowest n rolls or highest n rolls, respectively. So 2d20L1 means to roll two d20s and discard the lower. I.E advantage.
+            - A dice roll can be part of a mathematical expression, such as 1d4 +5.
+
+            **Example:**
+              `-roll 1d6` will roll a d6
+              `-roll (2d20L1) + 1d4 + 5` will roll 2d20s, discard the lower one, and add 1d4 and 5 to the result
+
+            *Full notation can be found here: https://xdice.readthedocs.io/en/latest/dice_notation.html*
+            """
+
+        #Todo:
+        # []: Make the special message system more robust and logical 
+
+
+        # Putting all the arguments in one string
+        # https://stackoverflow.com/questions/55665255/how-do-i-put-multiple-words-in-one-argument-discord-py
+        arg = " ".join(args[:])
+
+        try:
+            # Attempt to parse the user command and roll the dice
+            argument = xdice.Pattern(arg)
+            argument.compile()
+            roll = argument.roll()
+        except (SyntaxError, TypeError, ValueError):
+            raise commands.BadArgument()
+
+        # Add a special message if a user rolls a 20 or 1    
+        specialMessage = ""
+
+        # Loop over each dice roll and add it to the intermediate text
+        rollInformation = []
+        for score in roll.scores():
+            
+            scoreString = ""
+            
+            if type(score.detail) is int:
+                scoreString = scoreString + score.detail
+            elif type(score.detail) is list:
+                scoreString = scoreString + " + ".join(map(str,score.detail))
+            
+            if score.dropped == []:
+                pass
+            elif type(score.dropped) is int:
+                scoreString = scoreString + " ~~+ " +score.dropped + "~~"
+            elif type(score.dropped) is list:
+                scoreString = scoreString +  " ~~+ " + " + ".join(map(str,score.dropped)) + "~~"
+
+            #Check if we need a special message:
+            if "d20" in score.name:
+                if 1 in score.detail:
+                    specialMessage = "Aww, you rolled a natural 1"
+                elif 20 in score.detail:
+                    specialMessage = "Yay! You rolled a natural 20"
+
+
+            scoreString = "[{0}]".format(scoreString)
+            rollInformation.append(scoreString)
+
+        # Put spaces between the operators in the xdice template
+        formatString = argument.format_string
+        for i in ["+","-","/","*"]:
+            formatString = formatString.replace(i, " {0} ".format(i))
+        
+        #Format the intermediate text using the previous template
+        rolls = formatString.format(*rollInformation)
+
+        #Create the final message
+        msg = "{0}:\n> `{1}` = {2} = {3}".format(ctx.message.author.mention,arg,rolls,roll)
```

Comment:
```suggestion
        msg = f"{ctx.author.mention}:\n> `{arg}` = {rolls} = {roll}"
```

Better to use f-strings here.